grpc/federation/error.go

```diff
@@ -38,18 +40,53 @@ func GoWithRecover(eg *errgroup.Group, fn func() (any, error)) {
 	})
 }
 
+type ErrorWithLogAttrs struct {
+	err   error
+	attrs []slog.Attr
+}
+
+func (e *ErrorWithLogAttrs) Error() string {
+	return e.err.Error()
+}
+
+func (e *ErrorWithLogAttrs) Unwrap() error {
+	return e.err
+}
+
+func (e *ErrorWithLogAttrs) GRPCStatus() *grpcstatus.Status {
+	return grpcstatus.Convert(e.err)
+}
+
+func NewErrorWithLogAttrs(err error, attrs []slog.Attr) error {
+	if err == nil {
+		return nil
+	}
+	if errors.Is(err, &ErrorWithLogAttrs{}) {
```

Comment:
If I use this function, when the error is `ErrorWithLogAttrs,` I might want the function to merge `attrs` instead of simply returning the original error.

Reply:
I considered the case where an error is wrapped multiple times in a nested message, but I removed this code because such a case does not currently exist and it would be better to check the error outside of this function. 🙏